core/src/CompletionStream.ts

```diff
@@ -27,17 +27,33 @@ export class DefaultCompletionStream<PromptT = any> implements CompletionStream<
         const start = Date.now();
         const stream = await this.driver.requestCompletionStream(this.prompt, this.options);
 
+        let finish_reason: string = "";
+        let promptTokens: number = 0;
+        let resultTokens: number = -1;   //-1 Used as a trigger to see if it has been changed
         for await (const chunk of stream) {
             if (chunk) {
-                chunks.push(chunk);
-                yield chunk;
+                if (typeof chunk === 'string'){
+                    chunks.push(chunk);
+                    yield chunk;
+                }else{
+                    if(!(finish_reason == "stop" && !chunk.result)){    //Used to skip empty chunk after stop
+                        chunks.push(chunk.result);
+                        finish_reason = chunk.finish_reason ?? "";
+                        if(chunk.token_usage) {
+                            promptTokens = chunk.token_usage.prompt ?? 0;      //Tokens returned include prior parts of stream, 
+                            resultTokens = chunk.token_usage.result ?? 0;      //so overwrite rather than accumulate
+                        }
+                        yield chunk;
+                    }
+                }                
             }
         }
 
-        const content = chunks.join('');
+        //TODO: Confirm this is a true count for models/providers when streaming
```

Comment:
I would say to return -1 if we don't know the token count. Otherwise the result will be confusing for the user - if we return something that it is not what it pretends to be. The user can always estimate by itself the token number if -1 is returned.

Reply:
Returning chunk count as token count, removed entirely to avoid confusion.
Chunk count now passed to new property in completion, completion.chunks.